maro/simulator/scenarios/finance/common/trader.py

```diff
@@ -0,0 +1,252 @@
+import math
+from enum import Enum
+from collections import OrderedDict
+from maro.simulator.scenarios.finance.common.common import (
+    Action, OrderMode
+)
+
+
+class TradeConstrain(Enum):
+    min_buy_unit = "min_buy_unit"
+    min_sell_unit = "min_sell_unit"
+
+
+class Trader():
+    def __init__(self, trade_constrain: dict):
+        self._order_handlers = OrderedDict()
+        self._slippage_handler = None
+        self._commission_handlers = []
+        self._trade_constrain = OrderedDict()
+        for constrain in trade_constrain:
+            self._trade_constrain[constrain] = trade_constrain[constrain]
+
+        self._trade_number = 0
+
+    @property
+    def supported_orders(self) -> list:
+        return self._order_handlers.keys()
+
+    def order_handler_register(self, order_type: OrderMode, order_handler: callable):
+        self._order_handlers[order_type] = order_handler
+
+    def slippage_handler_register(self, slippage_handler: callable):
+        self._slippage_handler = slippage_handler
+
+    def commission_handler_register(self, commission_handler: callable):
+        self._commission_handlers.append(commission_handler)
+
+    def trade(self, order_action: Action, cur_data: dict, remaining_money: float):
+        # return stock, success, stock_price, number, tax
+        asset = order_action.item_index
+        is_success = False
+        is_trigger = True
+
+        deal_volume = abs(order_action.number)
+        is_buy = True if order_action.number >= 0 else False
+
+        deal_price = cur_data[order_action.item_index].opening_price
+        if 'deal_price' in self._trade_constrain:
+            if self._trade_constrain['deal_price'] == 'closing':
+                deal_price = cur_data[order_action.item_index].closing_price
+        commission_charge = 0
+
+        if order_action.order_mode not in self._order_handlers:
+            pass
+        else:
+            order_handler = self._order_handlers[order_action.order_mode]
+
+            if not order_handler.is_trigger(order_action, cur_data):
```

Comment:
all handler functions use only opening_price to adjust whether to trigger or not, not the deal_price

Reply:
done

Reply:
You disable the config of whether to use opening price or closing price?

Reply:
And by default, always using the opening price?